tfhe/src/integer/server_key/radix_parallel/tests_unsigned/test_aes.rs

```diff
@@ -0,0 +1,225 @@
+                | ((chunk[1] as u128) << 64)
+                | ((chunk[2] as u128) << 32)
+                | (chunk[3] as u128)
+        })
+        .collect()
+}
+fn sub_bytes(state: &mut [u8; 16]) {
+    for byte in state.iter_mut() {
+        *byte = S_BOX[*byte as usize];
+    }
+}
+fn shift_rows(state: &mut [u8; 16]) {
+    let original = *state;
+    state[1] = original[5];
+    state[5] = original[9];
+    state[9] = original[13];
+    state[13] = original[1];
+    state[2] = original[10];
+    state[6] = original[14];
+    state[10] = original[2];
+    state[14] = original[6];
+    state[3] = original[15];
+    state[7] = original[3];
+    state[11] = original[7];
+    state[15] = original[11];
+}
+fn gmul(mut a: u8, mut b: u8) -> u8 {
+    let mut p = 0;
+    for _ in 0..8 {
+        if (b & 1) != 0 {
+            p ^= a;
+        }
+        let hi_bit_set = (a & 0x80) != 0;
+        a <<= 1;
+        if hi_bit_set {
+            a ^= 0x1B;
+        }
+        b >>= 1;
+    }
+    p
+}
+fn mix_columns(state: &mut [u8; 16]) {
+    let original = *state;
+    for i in 0..4 {
+        let col = i * 4;
+        state[col] = gmul(original[col], 2)
+            ^ gmul(original[col + 1], 3)
+            ^ original[col + 2]
+            ^ original[col + 3];
+        state[col + 1] = original[col]
+            ^ gmul(original[col + 1], 2)
+            ^ gmul(original[col + 2], 3)
+            ^ original[col + 3];
+        state[col + 2] = original[col]
+            ^ original[col + 1]
+            ^ gmul(original[col + 2], 2)
+            ^ gmul(original[col + 3], 3);
+        state[col + 3] = gmul(original[col], 3)
+            ^ original[col + 1]
+            ^ original[col + 2]
+            ^ gmul(original[col + 3], 2);
+    }
+}
+fn add_round_key(state: &mut [u8; 16], round_key: u128) {
+    let key_bytes = round_key.to_be_bytes();
+    for i in 0..16 {
+        state[i] ^= key_bytes[i];
+    }
+}
+fn plain_aes_encrypt_block(block_bytes: &mut [u8; 16], expanded_keys: &[u128]) {
+    add_round_key(block_bytes, expanded_keys[0]);
+    for round_key in expanded_keys.iter().take(10).skip(1) {
+        sub_bytes(block_bytes);
+        shift_rows(block_bytes);
+        mix_columns(block_bytes);
+        add_round_key(block_bytes, *round_key);
+    }
+    sub_bytes(block_bytes);
+    shift_rows(block_bytes);
+    add_round_key(block_bytes, expanded_keys[10]);
+}
+fn plain_aes_ctr(num_aes_inputs: usize, iv: u128, key: u128) -> Vec<u128> {
+    let expanded_keys = plain_key_expansion(key);
+    let mut results = Vec::with_capacity(num_aes_inputs);
+    for i in 0..num_aes_inputs {
+        let counter_value = iv.wrapping_add(i as u128);
+        let mut block = counter_value.to_be_bytes();
+        plain_aes_encrypt_block(&mut block, &expanded_keys);
+        results.push(u128::from_be_bytes(block));
+    }
+    results
+}
+
+fn internal_aes_fixed_parallelism_test<P, E>(param: P, mut executor: E, num_aes_inputs: usize)
```

Comment:
Add some comments explaining what this test is testing. i.e., what is "fixed parallelism".